TextFormatter.java

```diff
@@ -19,49 +21,58 @@ public static void main(String[] args) {
 
   }
 
-  // Konstruktor
+
   public TextFormatter(int maxLineLength) {
 
     this.maxLineLength = maxLineLength;
 
   }
-  // Ausgabe
+
+
   public void print(String textA) {
 
-      String firstPart;
-      String remainder;
-   
-      while(textA.length() >= maxLineLength){
+    checkLengthAndPrint(textA);
+    System.out.println("Hier sollte der Text mit passendem Umbruch erscheinen.");
 
-        if (textA.charAt(maxLineLength - 1) == ' '){
-          firstPart = textA.substring(0, maxLineLength);
-          remainder = textA.substring(maxLineLength -1);
-          System.out.println(firstPart);
+  }
 
-        } else {
 
-          firstPart = textA.substring(0, maxLineLength - 1);
-          remainder = textA.substring(maxLineLength - 1);
+  public void checkLengthAndPrint(String textA){
+      
+      while(textA.length() >= maxLineLength){
 
-          if(textA.charAt(maxLineLength - 2) == ' ')  
-            System.out.println(firstPart );  
-          else
-            System.out.println(firstPart+ '-');
+        if (textA.charAt(maxLineLength - 1) == ' ')
+          printEntireWord(textA);
+         else 
+          printCutOffWord(textA);
           
-        }
-
         if(remainder.charAt(0) == ' ') remainder = remainder.substring(1);
-          
         textA = remainder;
         
       }
 
-      System.out.println(textA);
-      
-   
+        System.out.println(textA);
+
+  }
 
 
-    System.out.println("Hier sollte der Text mit passendem Umbruch erscheinen.");
+  private void printEntireWord(String textA){
+
+    firstPart = textA.substring(0, maxLineLength);
+    remainder = textA.substring(maxLineLength -1);
+    System.out.println(firstPart);
+
+  }
+
+
+  private void printCutOffWord(String textA){
+    firstPart = textA.substring(0, maxLineLength - 1);
+    remainder = textA.substring(maxLineLength - 1);
+
+    if(textA.charAt(maxLineLength - 2) == ' ')  
+       System.out.println(firstPart );  
+     else
+      System.out.println(firstPart+ '-');
   }
```

Comment:
Aktuell brechen Sie die Zeile um ohne auf die Wortstruktur zu achten. Sie sollten "Flattersatz" realisieren, d.h. das gesamte Wort, das nicht mehr in die Zeile passt, in die nächste übernehmen.